examples/rust/ex1-volume/src/main.rs

```diff
@@ -0,0 +1,429 @@
+    assert!(solution_order > 0);
+    let num_qpts = value_t!(command_line_args, "num_qpts", usize).unwrap();
+    assert!(num_qpts > 0);
+    let problem_size = value_t!(command_line_args, "problem_size", i64).unwrap();
+    let test = command_line_args.is_present("test");
+    let gallery = command_line_args.is_present("gallery");
+
+    // Summary output
+    if !test {
+        println!("Selected options: [command line option] : <current value>");
+        println!("    Ceed specification [-c] : {}", ceed_spec);
+        println!("    Mesh dimension     [-d] : {}", dim);
+        println!("    Mesh order         [-m] : {}", mesh_order);
+        println!("    Solution order     [-o] : {}", solution_order);
+        println!("    Num. 1D quadr. pts [-q] : {}", num_qpts);
+        println!("    Approx. # unknowns [-s] : {}", problem_size);
+        println!(
+            "    QFunction source   [-g] : {}",
+            if gallery { "gallery" } else { "user closure" }
+        );
+    }
+
+    // Initalize ceed context
+    let ceed = Ceed::init(ceed_spec);
+
+    // Mesh and solution bases
+    let basis_mesh =
+        ceed.basis_tensor_H1_Lagrange(dim, ncomp_x, mesh_order + 1, num_qpts, QuadMode::Gauss);
+    let basis_solution =
+        ceed.basis_tensor_H1_Lagrange(dim, ncomp_x, solution_order + 1, num_qpts, QuadMode::Gauss);
+
+    // Determine mesh size from approzimate problem size
+    let num_xyz = get_cartesian_mesh_size(dim, solution_order, problem_size);
+    if !test {
+        print!("Mesh size                   : nx = {}", num_xyz[0]);
+        if dim > 1 {
+            print!(", ny = {}", num_xyz[1]);
+        }
+        if dim > 2 {
+            print!(", nz = {}", num_xyz[2]);
+        }
+        print!("\n");
+    }
+
+    // Build ElemRestriction objects describing the mesh and solution discrete
+    // representations
+    let (restr_mesh, _) =
+        build_cartesian_restriction(&ceed, dim, num_xyz, mesh_order, ncomp_x, num_qpts);
+    let (restr_solution, restr_qdata) =
+        build_cartesian_restriction(&ceed, dim, num_xyz, solution_order, 1, num_qpts);
+    let mesh_size = restr_mesh.get_lvector_size() as usize;
+    let solution_size = restr_solution.get_lvector_size() as usize;
+    if !test {
+        println!("Number of mesh nodes        : {}", mesh_size / dim);
+        println!("Number of solution nodes    : {}", solution_size);
+    }
+
+    // Create a Vector with the mesh coordinates
+    let mut mesh_coords = set_cartesian_mesh_coords(&ceed, dim, num_xyz, mesh_order, mesh_size);
+
+    // Apply a transformation to the mesh coordinates
+    let exact_volume = transform_mesh_coordinates(dim, mesh_size, &mut mesh_coords);
+
+    // QFunction that builds the quadrature data for the mass operator
+    // -- QFunction from user closure
+    let build_mass =
+        |[jacobian, weights, ..]: QFunctionInputs, [qdata, ..]: QFunctionOutputs| -> i32 {
```

Comment:
```suggestion
        move |[jacobian, weights, ..]: QFunctionInputs, [qdata, ..]: QFunctionOutputs| -> i32 {
```
This is enough because `dim` is `Copy`. https://doc.rust-lang.org/std/keyword.move.html